### DISCUSSION TITLE
Building Prototypes

### DISCUSSION THREAD

**Discussion by Dakkaron** [Dakkaron/Fairberry#7] [General]
Hi All,
I have a few spare parts for the [bbq10kbd_breakout](https://github.com/arturo182/bbq10kbd_breakout) adapter. This is, by far, the most difficult part to solder.

The only other difficult part is removing the Arduino's USB port and soldering the USB C connector to it instead. But that's much simpler than the breakout board.

I have enough parts to make three adapter boards. I would solder them and send them per mail from Austria. I'd ask for the shipping cost.

Is anyone interested?

**Comment by spARTacusFromCB** [comment]
At least three folks from CB seemed to have indicated they are interested. Plus, at least a handful earlier in the CB thread that early indicated they'd probably be interested.

Is this a suitable alternate for the Arduino board?
--  CANADUINO® Pro Micro Atmega32u4, USB, 5V, 16MHz, 100% Compatible with Arduino https://www.amazon.ca/dp/B01N4TVIQX/ref=cm_sw_r_cp_apan_i_BNJRWWXZ4KZE6HJBG5KV?_encoding=UTF8&psc=1

**Reply by Dakkaron** [reply]
Yes, the board will work. Arduino is open source (and really simple) hardware, so there are tons of clones around, and almost all of them are fine.

I've read of people getting fake/broken devices from really cheap clone vendors, but I haven't gotten any so far, and if they have a decent rating on Amazon it will be fine.

**Reply by Dakkaron** [reply]
I am waiting with making new boards for now, since the custom PCB might be right around the corner.

**Comment by spARTacusFromCB** [comment]
Once the "Built-In" Arduino USB-A port is removed, do you normally short the two USB  points directly at / on the Arduino board to achieve the OTG, or do you normally do that within the wires leads arrangement going to the male USB-C connector? Presumably, doing it in the wire leads doesn't as much hinder the ability to still later update the chip and connecting to a PC? Or, maybe II am understanding that all wrong.

**Reply by Dakkaron** [reply]
I got myself USB OTG plugs. They have all the necessary stuff for OTG already in the plug and I only have to connect VCC, GND, D+ and D- to the board.

**Reply by spARTacusFromCB** [reply]
Cool.

**Comment by aaaaamir** [comment]
Hey @Dakkaron do you still have any spare parts kicking around? I'd like to give a go at this project for the Samsung s23 and avoid any difficult steps. Thanks!

**Reply by dumblob** [reply]
Hi @aaaaamir , I am glad to see someone coming across this! What @Dakkaron did and still does (!) is a great job and it really works (he has sent me a few pieces with full equipment and even valuable presents :wink:).

I am glad this repository is alive - see recent commits.

I for myself am planning to redo my current case design (below) to something smaller (because I noticed I really do not have any real world use cases for such a big screen).

![image](https://github.com/Dakkaron/Fairberry/assets/2451044/285ee7fc-73f2-46af-b6aa-eb1d31c88f6e)

But due to my personal life situation this is unfortunately a very low priority for now. The more I would like to encourage you to create something yours :wink:.

Vivat smartphone HW keyboards!

**Comment by theblcl** [comment]
Hey all,

I commented on the thread for flat design, but I'll take any design over no pkb. Does anyone have schematics for a pixel 7 or pixel 8? Or is there anything I can do to help get some schematics set up for people coming across this in the future?

**Comment by AmboCMH** [comment]
Hi,

Is this logic for the failure of qwerty keyboard sliders still correct when we see devices such as the CMF 1? Given it's weight and ability to secure a chassis on the back of the device. A short droid 1 style keyboard maybe makes sense now? A keyboard which is essentially 50% exposed under the device and 50% attached to the device? I am currently working on a prototype and will share progress. I do not however have the skills to incorporate the PCB required. I was thinking of purchasing a clicks for iPhone keyboard case and cannibalise this but the keyboard would be far too narrow. I considered the Gruber design for Arduino keyboard but would be difficult to incorporate. Your designs seem a good option, I do like the throughout USBc charging ability of the clicks keyboard. Let me know if you could help from a PCB programming perspective. Thank you

**Reply by Dakkaron** [reply]
Do you mean a portrait or landscape slider?

Both slider types share the problem that for a slider design the majority of the weight needs to be in the keyboard part, and that a USB connection is almost impossible.

If you go with a portrait slider (like the Blackberry Priv), the USB port would be directly on top of the keyboard, blocking access to it. So you need to go with bluetooth, need to add a separate battery and charging circuitry for the keyboard. You will need to charge your phone and keyboard separately each night and you will have to deal with the hassle of bluetooth problems.

It will also make it difficult to hold the device comfortably (because the keyboard will not only be 3-5mm behind the screen but more like 10-15mm). Sliders add thickness, so the whole assembly will probably add 10-15mm thickness to your phone.

Sourcing and adapting sliding mechanisms is also surprisingly difficult.

If you go for a landscape slider, you might be able to use USB, but you'll have the connector sticking out of the bottom of the phone. Apart from that, all the other issues still apply.

The only thing I could maybe see is a landscape slider if you can find an extremely lightweight phone.

But I too still dream of a good landscape slider, so I'd be happy to be proven wrong!

I'm still considering  getting a cheap phone, ripping off the screen, making a screen cable extension and putting a keyboard and slider assembly between body and screen. But I fear I'll never find time for that.
